ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/BerlinGasCalculator.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.mainnet;
+
+import static org.hyperledger.besu.ethereum.core.Address.BLS12_MAP_FP2_TO_G2;
+
+import org.hyperledger.besu.ethereum.core.Account;
+import org.hyperledger.besu.ethereum.core.Address;
+import org.hyperledger.besu.ethereum.core.Gas;
+import org.hyperledger.besu.ethereum.core.Wei;
+import org.hyperledger.besu.ethereum.vm.MessageFrame;
+
+import org.apache.tuweni.units.bigints.UInt256;
+
+public class BerlinGasCalculator extends IstanbulGasCalculator {
+
+  // new constants for EIP-2929
+  private static final Gas COLD_SLOAD_COST = Gas.of(2100);
+  private static final Gas COLD_ACCOUNT_ACCESS_COST = Gas.of(2600);
+  private static final Gas WARM_STORAGE_READ_COST = Gas.of(100);
+
+  // redefinitions for EIP-2929
+  private static final Gas SLOAD_GAS = WARM_STORAGE_READ_COST;
+  private static final Gas SSTORE_RESET_GAS = Gas.of(5000L).minus(COLD_SLOAD_COST);
+
+  // unchanged from Istanbul
```

Comment:
Why are we redefining these?

Reply:
We are copying in some formulas that require them.  The formulas we are bringing in are a mix of new values and old values, and the old values are private static.  I'm copying instead of upping the visibility to make it explicit what we are using and also to prevent accidental shadowing in future forks.